plugins/org.eclipse.n4js/src/org/eclipse/n4js/N4JSRuntimeModule.java

```diff
@@ -204,6 +206,14 @@
 		return ErrorAwareLinkingService.class;
 	}
 
+	/**
+	 * Bind the {@link ILocationInFileProvider} that is aware of derived elements.
+	 */
+	@Override
+	public Class<? extends ILocationInFileProvider> bindILocationInFileProvider() {
+		return N4JSLocationInFileProvider.class;
```

Comment:
This is for navigation to the correct line? Why isn't it needed in the ui case anymore?

Reply:
The UI injector "inherits" it from the runtime injector

Reply:
Should have been bound here right from the start

Reply:
I see.